File: 프로그래머스/Lv.2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    cnt = {}
    for i in tangerine:
        cnt[i] = cnt[i] + 1 if i in cnt else 1

    cnt = sorted(cnt.items(), key = lambda x: -x[1])
    
    ans = []
    while k > 0:
        x, y = cnt.pop(0)
        k -= y
        ans.append(x)
        
    answer = len(ans)    
    return answer

File: 프로그래머스/Lv.2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let answer = 0;
    
    let cnt = {}
    
    for (let i of tangerine) {
        i in cnt ? cnt[i] += 1 : cnt[i] = 1
    }
    // 1. for
    // 2. Object.entries
    
    let sortCnt = Object.entries(cnt)
    sortCnt.sort((a, b) => b[1] - a[1])
    
    let total = []
    while (k > 0) {
        let line = sortCnt.shift()
        k -= line[1]
        total.push(line[0])
    }
    
    answer = total.length
    return answer;
}